### PR TITLE
chore: update storybook

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2226]

### Description

Dependabot found security issues in our repo. One of critical issues is dependency of `merge-deep@3.0.2`

`merge-deep` is used by `storybook` and `jest-enviroment-puppeteer`
- storybook is taken care of in this PR
- puppeteer will has his own PR, when done, we can remove `merge-deep` from `resolutions`

`@storybook/react` also needed to be moved to resolutions, because of dependency `react-storybook-addon-chapters` which is dependant on the old storybook. It seems that it works with the new version just fine.

Some components' displayName needed to be updated to work correctly with the new storybook.

I have also fixed some exports of types to remove a lot of warnings in the console when running the storybook.

### How to test

- Storybook should run the same as before

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[FX-2226]: https://toptal-core.atlassian.net/browse/FX-2226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -142,6 +142,8 @@
     "lodash": "^4.17.21",
     "**/marksy/marked": "^0.6.0",
     "**/npx/npm": "^5.7.1",
-    "prettierx": "^0.17.0"
+    "prettierx": "^0.17.0",
```
Comment (denieler) [inline package.json:145]: we don't need prettierx anymore :) we use prettier

**Review by denieler — APPROVED** [review]

File: .storybook/components/PicassoBook/Chapter.tsx [hunk 2]
```diff
@@ -114,14 +114,14 @@ class Chapter extends Base {
     }
 
     if (!component.__docgenInfo) {
-      throw new Error(
+      console.error(
```
Comment (denieler) [inline .storybook/components/PicassoBook/Chapter.tsx:117]: I think we should throw an error here, so the build will fail on CI

File: .storybook/components/PicassoBook/Chapter.tsx [hunk 3]
```diff
@@ -114,14 +114,14 @@ class Chapter extends Base {
...
         `Failed to parse __docgenInfo for '${component.displayName}'`
       )
     }
 
-    const generatedDocumentation = documentationGenerator.transform(
-      component.__docgenInfo
-    )
+    const generatedDocumentation = component.__docgenInfo
+      ? documentationGenerator.transform(component.__docgenInfo)
+      : {}
```
Comment (Rootre) [inline .storybook/components/PicassoBook/Chapter.tsx:124]: If we throw an error on line 117, theres no need for this

**Review by Rootre — APPROVED** [review]

File: .storybook/components/PicassoBook/Chapter.tsx (hunk 2 above)
Comment (TomasSlama) [reply .storybook/components/PicassoBook/Chapter.tsx:117]: good point, it didn't come to my mind, that error would trigger a fail on CI. I will revert it.

**Review by dmaklygin1 — APPROVED** [review]
good work!

File: package.json (hunk 1 above)
Comment (TomasSlama) [reply package.json:145]: we didn't update davinci-syntax yet in Picasso. But I will do it tomorrow so I will not forget to remove this resolution

File: packages/picasso/src/Table/Table.tsx [hunk 4]
```diff
@@ -73,7 +73,7 @@ Table.defaultProps = {
   variant: 'bordered'
 }
 
-Table.displayName = 'PicassoTable'
+Table.displayName = 'Table'
```
Comment (OleksandrNechai) [inline packages/picasso/src/Table/Table.tsx:76]: These renamings are done for consistency?

Comment (TomasSlama) [reply packages/picasso/src/Table/Table.tsx:76]: No, when I updated the storybook to the latest, for components with this naming generating docs stoped working.

Comment (OleksandrNechai) [reply packages/picasso/src/Table/Table.tsx:76]: But why? :) 
The other problem is inconsistency though. I see many components are named without Picasso prefix.
Having prefix might be useful actually for users to see where are picasso components when they are debugging

**Pull request merged**